bts7960/bts7960.go

```diff
@@ -0,0 +1,77 @@
+package bts7960
+
+import (
+	"machine"
+	"time"
+)
+
+type Device struct {
+	lEn            machine.Pin
+	rEn            machine.Pin
+	lPwm           machine.Pin
+	rPwm           machine.Pin
+	rPwmCh, lPwmCh uint8
+	pwm            machine.PWM
+}
+
+// New returns a new motor driver.
+func New(lEn, rEn, lPwm, rPwm machine.Pin, pwm machine.PWM) *Device {
```

Comment:
Can you define and use a PWM interface instead of using `machine.PWM`? It is called differently on other chips.

For example, with the following interface:

```go
type PWM interface {
    Channel(pin machine.Pin) (channel uint8, err error)
    Set(channel uint8, value uint32)
    Top() uint32
}
```

```suggestion
func New(lEn, rEn, lPwm, rPwm machine.Pin, pwm PWM) *Device {
```